### PR TITLE
board: arm: nucleo_f207zg: add DAC support

### PR DESCRIPTION
**Pull request by ABOSTM**
board: arm: nucleo_f207zg: add DAC support
    
* Add DAC support to nucleo_f207zg
* Add fixture `dac_adc_loopback` to test it :  tests\drivers\dac\dac_loopback

### CODE REVIEW

**Review by galak — APPROVED** [review]

File: drivers/pinmux/stm32/pinmux_stm32f2.h [hunk 1]
```diff
@@ -50,6 +50,8 @@
 
 #define STM32F2_PINMUX_FUNC_PA4_ADC12_IN4	\
 	STM32_MODER_ANALOG_MODE
+#define STM32F2_PINMUX_FUNC_PA4_DAC_OUT1	\
+	STM32_MODER_ANALOG_MODE
```
Comment (martinjaeger) [inline drivers/pinmux/stm32/pinmux_stm32f2.h:54]: Please add also:
```
#define STM32F2_PINMUX_FUNC_PA5_DAC_OUT2 \
	STM32_MODER_ANALOG_MODE
```

**Review by martinjaeger — CHANGES_REQUESTED** [review]

File: drivers/pinmux/stm32/pinmux_stm32f2.h (hunk 1 above)
Comment (ABOSTM) [reply drivers/pinmux/stm32/pinmux_stm32f2.h:54]: Done

**Review by martinjaeger — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
dac_loopback test is failed on arduino_zero which declares support for both dac and adc but which is not supported in `tests/drivers/dac/dac_loopback/src/test_dac.c`, which seems a valid case.
Adding `depends on: dac adc` is not sufficient to solve this case, but unless adding a dac_loopback as a possible `supported` value, I don't see how this could be solved.

**Comment by ABOSTM** [comment]
Rebased.
@carlescufi @nashif @galak 
Could PR be merged for next release ?

**Comment by carlescufi** [comment]
> Rebased.
> @carlescufi @nashif @galak
> Could PR be merged for next release ?

Done

**Pull request merged**